### PR TITLE
refactor(event): admin

### PR DESCRIPTION
**Pull request by olros**
Lint, test and build
## Description

Comments/issues/screenshots:
- The signup-FAQ somehow got removed in a pr, so its reintroduced here
- Should use global react-hook-form components (see #197 ) when created
- SignUp-switch uses useState as it was unpredictable with react-hook-form when switching between events. Didn't try to use the markdown-editor with react-hook-form
- No forms yet, but this should make it easier
- New markdown-editor as in #207 
- Also relevant for #243 and #255 

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] The PR includes a `closes #issueID`
- [ ] The PR includes a picture showing visual changes

## Typescript convert checklist

- [x] All relative imports has been converted to absolute imports
- [x] All HOC's have been replaced with Hooks
- [x] Uses theme.spacing() instead of custom padding
- [x] Uses theme.breakpoints instead of custom media queries

### CODE REVIEW

File: src/components/layout/SidebarList.tsx [hunk 1]
```diff
@@ -1,172 +1,144 @@
...
+                </ListItem>
+              ))}
+            </List>
           </>
         )}
-      </Grid>
-    </div>
+      </Drawer>
+      <Hidden lgUp>
+        <Zoom in={!mobileOpen} style={{ transitionDelay: `${!mobileOpen ? transitionDuration.exit : 0}ms` }} timeout={transitionDuration} unmountOnExit>
```
Comment (Zenjjim) [inline src/components/layout/SidebarList.tsx:135]:
```suggestion
        <Zoom in={!mobileOpen} style={{ transitionDelay: `${mobileOpen ? 0 : transitionDuration.exit }ms` }} timeout={transitionDuration} unmountOnExit>
```

File: src/containers/EventAdministration/components/EventParticipants.tsx [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+  const showSnackbar = useSnackbar();
+  const [event, setEvent] = useState<Event | null>(null);
+  const [showOnlyNotAttended, setShowOnlyNotAttended] = useState(false);
+  const [showEmails, setShowEmails] = useState(false);
+  const [participants, setParticipants] = useState<Array<Registration>>([]);
+
+  useEffect(() => {
+    getEventRegistrations(eventId)
+      .then((result) => setParticipants(result))
+      .catch(() => {});
```
Comment (Zenjjim) [inline src/containers/EventAdministration/components/EventParticipants.tsx:90]: fjern denne?

File: src/containers/EventAdministration/components/EventParticipants.tsx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+  }, [eventId, getEventRegistrations]);
+
+  useEffect(() => {
+    getEventById(eventId)
+      .then((result) => setEvent(result))
+      .catch(() => {});
```
Comment (Zenjjim) [inline src/containers/EventAdministration/components/EventParticipants.tsx:96]: fjern

File: src/containers/EventAdministration/components/EventParticipants.tsx [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+        setParticipants(newParticipants);
+        showSnackbar('Endringen var vellykket', 'success');
+      })
+      .catch((error) => showSnackbar(error.detail, 'error'));
+  };
+
+  const getOnWaitlist = useMemo(() => participants.filter((user) => user.is_on_wait), [participants]);
+  const getAttending = useMemo(() => participants.filter((user) => !user.is_on_wait), [participants]);
+
+  const printParticipants = (onWaitlist = false, onlyNotAttended = false) => {
```
Comment (Zenjjim) [inline src/containers/EventAdministration/components/EventParticipants.tsx:124]: Denne burde gjøres til react

File: src/containers/EventAdministration/components/EventParticipants.tsx [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+          {event?.title || 'Laster...'}
+        </Typography>
+        <div className={classes.numbers}>
+          <Typography className={classes.lightText}>Antall påmeldte: {getAttending.length}</Typography>
+          <Typography className={classes.lightText}>Antall på venteliste: {getOnWaitlist.length}</Typography>
+        </div>
+      </div>
+      <Divider />
+      <div className={classes.content}>
+        {getAttending.length > 0 && (
```
Comment (Zenjjim) [inline src/containers/EventAdministration/components/EventParticipants.tsx:185]:
```suggestion
        {getAttending.length && (
```

File: src/containers/EventAdministration/components/EventRegistrationPriorities.tsx [hunk 6]
```diff
@@ -50,44 +51,46 @@ const styles = (theme) => ({
...
+const EventRegistrationPriorities = ({ priorities, setPriorities }: EventRegistrationPrioritiesProps) => {
+  const classes = useStyles();
 
-  const handlePriorityChange = (userClass, userStudy) => () => {
-    if (event.registration_priorities.some((item) => item.user_class === userClass && item.user_study === userStudy)) {
-      const newArray = event.registration_priorities.filter((item) => !(item.user_class === userClass && item.user_study === userStudy));
-      setEvent({ ...event, registration_priorities: newArray });
+  const handlePriorityChange = (userClass: UserClass, userStudy: UserStudy) => () => {
+    if (priorities.some((item) => item.user_class === userClass && item.user_study === userStudy)) {
+      const newArray = priorities.filter((item) => !(item.user_class === userClass && item.user_study === userStudy));
```
Comment (Zenjjim) [inline src/containers/EventAdministration/components/EventRegistrationPriorities.tsx:66]: bedre navn

File: src/containers/EventAdministration/components/EventStatistics.tsx [hunk 7]
```diff
@@ -38,25 +38,29 @@ const styles = (theme) => ({
...
+const EventStatistics = ({ registrations }: EventStatisticsProps) => {
+  const classes = useStyles();
 
-  const attendedNo = (attended) => participants.filter((x) => x.has_attended === attended).length;
+  const attendedNumber = useMemo(() => registrations.filter((x) => x.has_attended).length, [registrations]);
 
-  const classNo = (userClass) => {
-    const no = participants.filter((x) => x.user_info.user_class === userClass).length;
+  const classNo = (userClass: number) => {
+    const no = registrations.filter((x) => x.user_info.user_class === userClass).length;
```
Comment (Zenjjim) [inline src/containers/EventAdministration/components/EventStatistics.tsx:53]:
```suggestion
    const no = registrations.filter((registration) => registration.user_info.user_class === userClass).length;
```

File: src/containers/EventAdministration/components/EventStatistics.tsx [hunk 8]
```diff
@@ -38,25 +38,29 @@ const styles = (theme) => ({
...
     return no > 0 ? no : '0';
   };
-  const studyNo = (userStudy) => {
-    const no = participants.filter((x) => x.user_info.user_study === userStudy).length;
+  const studyNo = (userStudy: number) => {
+    const no = registrations.filter((x) => x.user_info.user_study === userStudy).length;
```
Comment (Zenjjim) [inline src/containers/EventAdministration/components/EventStatistics.tsx:57]:
```suggestion
    const no = registrations.filter((registration) => registration.user_info.user_study === userStudy).length;
```

File: src/containers/EventAdministration/components/EventStatistics.tsx [hunk 9]
```diff
@@ -38,25 +38,29 @@ const styles = (theme) => ({
...
     return no > 0 ? no : '0';
   };
 
   return (
-    <React.Fragment>
-      <Typography className={classes.lightText}>Ankommet: {attendedNo(true) + ' av ' + participants.length} påmeldte</Typography>
+    <>
+      <Typography className={classes.lightText}>Ankommet: {attendedNumber + ' av ' + registrations.length} påmeldte</Typography>
```
Comment (Zenjjim) [inline src/containers/EventAdministration/components/EventStatistics.tsx:63]:
```suggestion
      <Typography className={classes.lightText}>Ankommet: { `${attendedNumber} av ${registrations.length}`} påmeldte</Typography>
```

File: src/containers/EventAdministration/components/Participant.tsx [hunk 10]
```diff
@@ -61,60 +60,67 @@ const style = (theme) => ({
...
+      <Dialog
+        confirmText='Ja, jeg er sikker'
+        contentText={`Er du sikker på at du vil fjerne ${userInfo.first_name} ${userInfo.last_name} fra arrangementet?`}
+        onClose={() => setShowModal(false)}
+        onConfirm={deleteHandler}
+        open={showModal}
+        titleText='Er du sikker?'
+      />
       <div className={classes.userName}>
         <Typography>{userInfo.first_name + ' ' + userInfo.last_name}</Typography>
```
Comment (Zenjjim) [inline src/containers/EventAdministration/components/Participant.tsx:103]:
```suggestion
        <Typography>{`${userInfo.first_name} ${userInfo.last_name}`}</Typography>
```

File: src/containers/EventAdministration/index.tsx [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+
+  const fetchExpired = () => {
+    if (isLoading) {
+      return;
+    }
+    setIsLoading(true);
+
+    getExpiredEvents()
+      .then((data) => setExpiredItems(data.results || data || []))
+      .catch(() => {})
```
Comment (Zenjjim) [inline src/containers/EventAdministration/index.tsx:85]: remove

File: src/containers/EventAdministration/index.tsx [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+      />
+      <div className={classes.root}>
+        <div className={classes.content}>
+          <Typography className={classes.header} variant='h2'>
+            {eventId ? 'Endre arrangement' : 'Nytt arrangement'}
+          </Typography>
+          <Tabs indicatorColor='primary' onChange={(e, newTab) => setTab(newTab)} textColor='primary' value={tab} variant='scrollable'>
+            <Tab id='0' label={eventId ? 'Endre' : 'Skriv'} />
+            {eventId && <Tab id='1' label='Deltagere' />}
+            {eventId && <Tab component={Link} id='2' label='Registrering' to={`${URLS.events}${eventId}/registrering/`} />}
```
Comment (Zenjjim) [inline src/containers/EventAdministration/index.tsx:124]:
```suggestion
            {eventId && <Tab component={Link} id='2' label='Registrering' to={`${URLS.events}/${eventId}/registrering/`} />}
```

File: src/containers/SignUp/index.js [hunk 13]
```diff
@@ -286,13 +290,34 @@ function SignUp(props) {
...
                 <Button className={classes.button} color='primary' disabled={isLoading} type='submit' variant='contained'>
                   Opprett bruker
                 </Button>
                 <Button className={classes.button} color='primary' component={Link} disabled={isLoading} to={URLS.login}>
                   Logg inn
                 </Button>
+                <Button
+                  className={classes.button}
+                  color='primary'
+                  endIcon={!faqOpen ? <ExpandMoreIcon /> : <ExpandLessIcon />}
```
Comment (Zenjjim) [inline src/containers/SignUp/index.js:305]:
```suggestion
                  endIcon={faqOpen ? <ExpandLessIcon /> : <ExpandMoreIcon /> }
```

File: src/containers/SignUp/index.js [hunk 14]
```diff
@@ -286,13 +290,34 @@ function SignUp(props) {
...
+                  onClick={() => setFaqOpen((oldState) => !oldState)}
+                  variant='outlined'>
+                  Hvorfor må vi godkjenne deg?
+                </Button>
+                <Collapse in={faqOpen}>
+                  <Typography className={classes.button} variant='body2'>
+                    For å unngå at vi får mange brukere som ikke er reelle TIHLDE-medlemmer, må vi aktivere nye brukere før de får logge inn. Det kan ta noen
+                    timer før noen av oss i Index eller HS får verifisert brukeren din. Hvis det haster eller tar mer enn 24 timer kan du sende en melding til
+                    TIHLDE på{' '}
```
Comment (Zenjjim) [inline src/containers/SignUp/index.js:314]:
```suggestion
                    TIHLDE på
```

File: src/index.tsx [hunk 15]
```diff
@@ -116,6 +116,62 @@ const requireAuth = (OriginalComponent: React.ReactElement, accessGroups: Array<
...
+      .then((user) => {
+        if (isSubscribed) {
+          groups.forEach((group) => {
+            const userGroups = user?.groups || [];
+            const regex = new RegExp(userGroups.join('|'), 'i');
+            if (regex.test(group)) {
+              setAllowAccess(true);
+            }
+          });
+          if (isAuthenticated() && groups.length === 0) {
```
Comment (Zenjjim) [inline src/index.tsx:145]:
```suggestion
          if (isAuthenticated() && groups.length) {
```

File: src/index.tsx [hunk 16]
```diff
@@ -116,6 +116,62 @@ const requireAuth = (OriginalComponent: React.ReactElement, accessGroups: Array<
...
+            setAllowAccess(true);
+          }
+          setIsLoading(false);
+        }
+      })
+      .catch(() => {});
```
Comment (Zenjjim) [inline src/index.tsx:151]: fjern

**Review by Zenjjim — CHANGES_REQUESTED** [review]
HeHe

File: src/containers/EventAdministration/index.tsx (hunk 12 above)
Comment (olros) [reply src/containers/EventAdministration/index.tsx:124]: Dette er feil. `URLS.events` = `/arrangementer/`

**Review by Zenjjim — APPROVED** [review]

**Pull request merged**